ui/components/WorkloadOnboardingHints.tsx

```diff
@@ -0,0 +1,271 @@
+    sourceCheck();
+
+    let interval;
+    if (!sourceCreated) {
+      // Check every 5 seconds and clear if source is found
+      interval = setInterval(() => {
+        sourceCheck();
+      }, 5000);
+    } else {
+      clearInterval(interval);
+    }
+
+    return () => {
+      clearInterval(interval);
+    };
+  }, [sourceCreated]);
+
+  React.useEffect(() => {
+    if (!sourceCreated) {
+      // Don't do anything if a source isn't created yet
+      return;
+    }
+    kustomizationCheck();
+
+    let interval;
+    if (!kustomizationCreated) {
+      interval = setInterval(() => {
+        if (kustomizationCreated) {
+          clearInterval(interval);
+        } else {
+          kustomizationCheck();
+        }
+      }, 5000);
+    } else {
+      clearInterval(interval);
+    }
+
+    return () => {
+      clearInterval(interval);
+    };
+  }, [sourceCreated, kustomizationCreated]);
+
+  return (
+    <div className={className}>
+      <Box paddingBottom={4}>
+        <FormControl variant="outlined">
+          <Flex wide>
+            <Box marginBottom={2} marginRight={4}>
+              <TextField
+                variant="outlined"
+                label="Name"
+                value={formValues.name}
+                onChange={(e) =>
+                  setFormValues({ ...formValues, name: e.target.value })
+                }
+              />
+            </Box>
+            <Box marginRight={4}>
+              <StyledSelect
+                onChange={(e) => {
+                  setFormValues({
+                    ...formValues,
+                    sourceType: e.target.value as SourceType,
+                  });
+                }}
+                value={formValues.sourceType}
+              >
+                <MenuItem value={SourceType.Git}>{SourceType.Git}</MenuItem>
+                <MenuItem value={SourceType.Bucket}>
+                  {SourceType.Bucket}
+                </MenuItem>
+                <MenuItem value={SourceType.Helm}>{SourceType.Helm}</MenuItem>
+              </StyledSelect>
+            </Box>
+          </Flex>
+          <Flex>
+            <Box marginRight={4}>
+              <TextField
+                variant="outlined"
+                label="Source URL"
+                value={formValues.sourceUrl}
+                onChange={(e) =>
+                  setFormValues({ ...formValues, sourceUrl: e.target.value })
+                }
+              />
+            </Box>
+            <Box marginRight={4}>
+              <TextField
+                variant="outlined"
+                label="Clusters YAML Path"
+                value={formValues.yamlPath}
+                onChange={(e) =>
+                  setFormValues({ ...formValues, yamlPath: e.target.value })
+                }
+              />
+            </Box>
+          </Flex>
+        </FormControl>
+      </Box>
+
+      <div>
+        <Flex align>
+          <Number style={{ backgroundColor: sourceCreated ? "green" : "gray" }}>
```

Comment:
This `gray` is a bit to dark, not enough contrast with the black '1' – really appreciate all of the instructional pages like this and the color changing follow along.

<img width="1440" alt="Screenshot 2021-03-13 at 22 42 25" src="https://user-images.githubusercontent.com/92439/111048378-8f82e300-844d-11eb-801a-0ad0c7f3078c.png">

Reply:
Yeah that color will be replaced very soon

Reply:
We assume here the workload comes from a public repo, but I think in most cases it comes from a private repo. I would add `flux create secret git <SSH secret>` and `flux create source git --secretRef=<SSH secret>`.

Reply:
> We assume here the workload comes from a public repo, but I think in most cases it comes from a private repo. I would add `flux create secret git <SSH secret>` and `flux create source git --secretRef=<SSH secret>`.

:+1: This is a naive implementation for now.

@stefanprodan Do you think we should try to pre-fill the "Clusters YAML Path", perhaps by looking at the last created Source or Kustomization? My thinking is that it will be hard to accurately pre-fill the value with a one-size-fits-all default.

Reply:
Converted to issue #14

Reply:
> My thinking is that it will be hard to accurately pre-fill the value with a one-size-fits-all default.

For sure, I think it's best to add links to the docs for each command, this way people can see more advanced examples like: https://toolkit.fluxcd.io/components/source/gitrepositories/#spec-examples